### DISCUSSION TITLE
deformable solids attached to rigid bodies

### DISCUSSION THREAD

**Discussion by soldierofhell** [InteractiveComputerGraphics/SPlisHSPlasH#255] [Q&A]
Hi, I wonder if we can simulate deformable solids attached to rigid bodies (attraction force of boundary particles toward rigid body?), something similar to this example from _Corotated SPH for deformable solids_
![Screenshot from 2023-06-20 20-11-02](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/45559276/91ed2a03-2de3-4b9c-b77c-cef2993ff6b9)

Regards,

**Comment by janbender** [comment]
Hi, 

you can fix particles. This you can define by defining a bounding box in the scene file (just recognized that these fields are missing in the current documentation):

* fixedBoxMin (vec3): Minimum point of box of which contains the fixed particles.
* fixedBoxMax (vec3): Maximum point of box of which contains the fixed particles.

All particles which are initially in this box are then fixed and cannot move. 

Hope that helps!

**Comment by soldierofhell** [comment]
Thank you for your answer, but actually I'd like to achieve a bit more, I'd like to attach deformable object to rigid body, so I could actuate rigid body and thus steer the deformable.. in the water. Even if this is currently not possible, could you guide me what has to be potentially added? Regards,

**Comment by janbender** [comment]
This is not supported at the moment. You could achieve this by defining a constraint: 
C = a - b = 0, 
where a is a connector point in the rigid body and b is the position of a particle in the deformable solid. Typically, a = R * a_local + t, where R and t are rotation and translation of the rigid body, and a_local defines the position of the connector point in local coordinates. 

With this constraint you could either attach the solid by penalty forces or use a position-based approach (PBD). In both cases you could compute additional forces which you consider in the time integration.
